### PR TITLE
Full text indexing is not returning expected results

### PR DESCRIPTION
**Pull request by stuartf**
I uploaded a random paper with the title "Personalized Web Search" (not the subject of the paper) and a paper about "Personalized Web Search" with an unrelated name. When searching for "Personalized Web Search", only the first document is returned. The second document isn't in the list at all, suggesting that there is a problem with the full text indexing.

### CODE REVIEW

**Comment by simong** [comment]
@stuartf: Is this ready for review?

**Comment by stuartf** [comment]
Not ready for review yet, for some reason this breaks content search queries (an error gets logged in elasticsearch). I'm trying to track down what's going on there, and am open to suggestions.

**Comment by stuartf** [comment]
I logged a working search from master, and after getting the formatting all matched up the only difference is that it doesn't have that `"fields": "*"` at the end.

**Comment by stuartf** [comment]
Well, re-running it on my branch I'm no longer seeing the error and `"fields": "*"` isn't showing in the log there either.

File: node_modules/oae-preview-processor/lib/processors/file/pdf.js [hunk 1]
```diff
@@ -308,3 +313,32 @@ var _convertToPNG = function(ctx, input, output, width, callback) {
...
+ *
+ * @param  {PreviewContext}      ctx             The preview context associated to this file
+ * @param  {String}              input           The PDF file to convert
+ * @param  {String}              output          The path where the txt file should be written
+ * @param  {Function}            callback        Standard callback function
+ * @param  {Object}              callback.err    An error that occurred, if any
+ * @api private
+ */
+var _convertToText = function(ctx, input, output, callback) {
+    var cmd = util.format('pdftotext -nopgbrk -q "%s" "%s"', input, output);
```
Comment (simong) [inline node_modules/oae-preview-processor/lib/processors/file/pdf.js]: It'd be nice to have the following files:
 - plain.txt  - contains the plain text data for the entire pdf and can be used for search indexing
 - page.1.txt - Can be used by the UI as an accessibility hint for the first page
 - page.2.txt - ...
 - ..
 - page.n.txt

File: config.js [hunk 2]
```diff
@@ -191,7 +191,6 @@ config.search = {
                     },
                     'html_content': {
```
Comment (simong) [inline config.js]: Now that we'll be indexing plain text, do we still need the html_content analyzer?

File: node_modules/oae-preview-processor/lib/processors/file/pdf.js [hunk 3]
```diff
@@ -308,3 +313,32 @@ var _convertToPNG = function(ctx, input, output, width, callback) {
...
+     // Execute the command
+    log().trace({'contentId': ctx.contentId}, 'Executing %s', cmd);
+    var options = {
+        'timeout': _pdftkTimeout,
+        'env': _.defaults({'OMP_NUM_THREADS': 2}, process.env)
+    };
+    exec(cmd, options, function (err, stdout, stderr) {
+        if (err) {
+            log().error({'err': err, 'contentId': ctx.contentId, 'stdout': stdout, 'stderr': stderr}, 'Could not convert a PDF page to plain text');
+            return callback({'code': 500, 'msg': 'Could not convert a PDF page to a PNG'});
```
Comment (simong) [inline node_modules/oae-preview-processor/lib/processors/file/pdf.js]: The error messages are wrong here

**Comment by simong** [comment]
The tests are passing locally and I'm not seeing anything regarding search failures, but there are quite a bit of test failures on Travis CI in the PDF previews area though

File: config.js (hunk 2 above)
Comment (stuartf) [inline config.js]: I think we still need an analyzer, but maybe it should be renamed.

File: node_modules/oae-content/lib/search.js [hunk 4]
```diff
@@ -226,57 +226,34 @@ var _produceContentBodyDocuments = function(resources, callback, _documents) {
...
+                fs.readFile(file.path, function(err, data) {
                     if (err) {
-                        return errorCallback(err);
+                        callback(err);
+                    } else {
+                        var childDoc = SearchUtil.createChildSearchDocument(ContentConstants.search.MAPPING_CONTENT_BODY, resource.id, {'content_body': data});
+
+                        // Add it and move on to the next document
+                        _documents.push(childDoc);
+                        return _produceContentBodyDocuments(resources, callback, _documents);
```
Comment (simong) [inline node_modules/oae-content/lib/search.js]: Can we do:

```javascript
if (!err) {
    var childDoc = SearchUtil.createChildSearchDocument(ContentConstants.search.MAPPING_CONTENT_BODY, resource.id, {'content_body': data});
    _documents.push(childDoc);
}

// In all cases, the file should be removed again
fs.unlink(file.path, function(err) {
    if (err) {
        log().error({'err': err}, 'Could not remove the plain text file when indexing content');
    }

    // Move on to the next file
    _produceContentBodyDocuments(resources, callback, _documents);
});
```

File: node_modules/oae-preview-processor/lib/processors/file/pdf.js [hunk 5]
```diff
@@ -308,3 +313,48 @@ var _convertToPNG = function(ctx, input, output, width, callback) {
...
+
+        // Save individual page copies of the plain text as well
+        fs.readFile(output, {'encoding': 'utf8'}, function(err, text) {
+            if (err) {
+                log().error({'err': err, 'contentId': ctx.contentId}, 'Could not read the generated plain text');
+                return callback({'code': 500, 'msg': 'Could not read the generated plain text'});
+            }
+            var pages = text.split('\f');
+            _.each(pages, function(page, index) {
+                var done = _.after(pages.length, callback);
```
Comment (simong) [inline node_modules/oae-preview-processor/lib/processors/file/pdf.js]: This needs to be moved outside of the _.each

File: node_modules/oae-content/lib/search.js [hunk 6]
```diff
@@ -226,57 +226,34 @@ var _produceContentBodyDocuments = function(resources, callback, _documents) {
...
+                    return callback(err);
                 }
 
-                ContentUtil.getStorageBackend(null, preview.uri).get(tenantAlias, preview.uri, function(err, file) {
+                fs.readFile(file.path, function(err, data) {
                     if (err) {
-                        return errorCallback(err);
+                        callback(err);
+                    } else {
+                        var childDoc = SearchUtil.createChildSearchDocument(ContentConstants.search.MAPPING_CONTENT_BODY, resource.id, {'content_body': data});
```
Comment (simong) [inline node_modules/oae-content/lib/search.js]: `data.toString()` as data is a Buffer instance

**Comment by simong** [comment]
There are still a couple of test failures on Travis. Can you push this to a branch in the `oaeproject` repo and resend a PR? That way, Travis will be able to upload the logs to S3.

**Comment by stuartf** [comment]
new PR made at #1103

**Comment by simong** [comment]
Closing in favor of #1103

**Pull request closed without merging**